### PR TITLE
feat: add collapse button for route labels

### PR DESCRIPTION
**Pull request by andonimihai**
Close https://github.com/roataway/roataway-web/issues/132
image
image
image
image

### CODE REVIEW

**Comment by roata** [comment]
Thank you. I have just a tweak - the color of the collapse arrows should be of the same style as the "x". At the moment it is too dark and attracts unnecessary attention.

I know it was black in the original sketch, at that stage I merely conveyed the main idea, without thinking about specific colour representations.

File: src/components/colors-legend.components.tsx [hunk 1]
```diff
@@ -19,11 +21,16 @@ import { useSelectedRoutes } from '../selected-routes.context'
...
   const { routes, setRoutes } = useSelectedRoutes()
+  const [isLegendCollapsed, setLegendCollapseState] = useState<boolean>(false)
 
   if (!allRoutes || Object.keys(colors).length === 0) {
     return null
   }
 
+  const collapseLegend = () => {
+    setLegendCollapseState(!isLegendCollapsed)
+  }
```
Comment (strdr4605) [inline src/components/colors-legend.components.tsx:32]: This function implies inversion of previous state so `toggleLegendCollapseState` is a better name.
```suggestion
  const toggleLegendCollapseState = () => {
    setLegendCollapseState(s => !s)
  }
```

File: src/components/colors-legend.components.tsx [hunk 2]
```diff
@@ -32,25 +39,33 @@ const ColorsLegend: FC<MapControlProps> = ({ position = 'topright' }) => {
...
+  const collapseButtonIconClasses = `${isLegendCollapsed ? legendClasses.rotatedToggleIcon : ''}`
   return (
     <MapControl position={position}>
       <Paper className={legendClasses.legendContainer} elevation={3}>
+        <div className={legendClasses.collapseWrapper}>
+          <Button className={legendClasses.collapseButton} onClick={collapseLegend}>
+            <DoubleArrowRoundedIcon className={collapseButtonIconClasses} />
+          </Button>
+        </div>
+
```
Comment (strdr4605) [inline src/components/colors-legend.components.tsx:68]: No need for new line

File: src/components/colors-legend.components.tsx [hunk 3]
```diff
@@ -32,25 +39,33 @@ const ColorsLegend: FC<MapControlProps> = ({ position = 'topright' }) => {
...
 
+  const collapseButtonIconClasses = `${isLegendCollapsed ? legendClasses.rotatedToggleIcon : ''}`
   return (
     <MapControl position={position}>
       <Paper className={legendClasses.legendContainer} elevation={3}>
+        <div className={legendClasses.collapseWrapper}>
+          <Button className={legendClasses.collapseButton} onClick={collapseLegend}>
+            <DoubleArrowRoundedIcon className={collapseButtonIconClasses} />
+          </Button>
+        </div>
```
Comment (strdr4605) [inline src/components/colors-legend.components.tsx:67]: Maybe have a separate component like `RemoveRouteButton` ?

File: src/components/colors-legend.module.scss [hunk 4]
```diff
@@ -14,13 +14,50 @@
...
   overflow-y: auto;
+  padding-left: 40px;
+}
+
+.collapseWrapper {
+  position: absolute;
+  left: 0;
+  top: 0;
+  height: 100%;
+  width: 40px;
```
Comment (strdr4605) [inline src/components/colors-legend.module.scss:52]: CSS is my weakness and a layout like this confuses me.
Is it possible to not use `padding-left` and `position: absolute;` here?

Maybe:

```css
.legendContainer {
  display: flex;
  align-items: center;
```

File: src/components/colors-legend.components.tsx (hunk 3 above)
Comment (andonimihai) [reply src/components/colors-legend.components.tsx:67]: @strdr4605 Did you mean `ToggleRoutesButton` instead of `RemoveRouteButton`?

Comment (strdr4605) [reply src/components/colors-legend.components.tsx:67]: Yes, I just wrote “like remove...”

File: src/components/colors-legend.module.scss (hunk 4 above)
Comment (andonimihai) [reply src/components/colors-legend.module.scss:52]: trying this way, I get stuck with 100% height for the child when the parent doesn't have a fixed height but only max-height. Do you have some suggestions on how to fix it?

Comment (strdr4605) [reply src/components/colors-legend.module.scss:52]: Strange !!!!
```css
display: flex;
align-items: center;
```
should do the work.
Maybe there are some issues with the collapseWrapper. Did you also apply `align-items: center;` for it?

**Review by strdr4605 — APPROVED** [review]

**Comment by roata** [comment]
@andonimihai Thank you for your help, and @strdr4605 thanks for suggesting improvements.

**Pull request merged**